src/components/tabs/ScatterPlotTab.tsx

```diff
@@ -111,11 +107,21 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
 
   // Debounce changes to the dataset to prevent noticeably blocking the UI thread with a re-render.
   // Show the loading spinner right away, but don't initiate the state update + render until the debounce has settled.
-  // TODO: React can merge multiple transitions into one, but it's not guaranteed behavior. Can we merge the transitions?
-  const dataset = useTransitionedDebounce(props.dataset, startTransition, () => setIsRendering(true), 500);
-  const isPlaying = useTransitionedDebounce(props.isPlaying, startTransition, () => setIsRendering(true), 5);
-  const selectedTrack = useTransitionedDebounce(props.selectedTrack, startTransition, () => setIsRendering(true), 0);
-  const currentFrame = useTransitionedDebounce(props.currentFrame, startTransition, () => setIsRendering(true), 0);
+  const { selectedTrack, currentFrame, colorRamp, categoricalPalette, selectedFeatureName, isPlaying, viewerConfig } =
+    props;
+  const dataset = useDebounce(props.dataset, 500);
+  const colorRampMin = useDebounce(props.colorRampMin, 100);
+  const colorRampMax = useDebounce(props.colorRampMax, 100);
+
+  const isDebouncePending =
+    dataset !== props.dataset || colorRampMin !== props.colorRampMin || colorRampMax !== props.colorRampMax;
+
+  // Signal that the plot is stalled when playback starts.
+  useEffect(() => {
+    if (isPlaying) {
+      setIsRendering(true);
+    }
+  }, [isPlaying]);
```

Comment:
Would it work to add `isPlaying` to the expression below that determines whether the loading spinner is shown, rather than updating state here?